### PR TITLE
Ensuring cookie_secure is true

### PR DESCRIPTION
**Pull request by SeemaMenon**
## Purpose
https://opgtransform.atlassian.net/browse/UML-125

Fixes UML-125

## Approach
Have changed code to pass an environment variable to set cookie_secure to false for local environments using docker-compose, and default to true otherwise.

## Learning
_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/app/config/autoload/envs.global.php [hunk 1]
```diff
@@ -58,7 +58,8 @@
         // Indicates that the cookie should only be transmitted over a
         // secure HTTPS connection from the client. When set to TRUE, the
         // cookie will only be set if a secure connection exists.
-        'cookie_secure' => false,
+//        'cookie_secure' => false,
+        'cookie_secure' => getenv('COOKIE_SECURE') ?: true,
```
Comment (hawx) [inline service-front/app/config/autoload/envs.global.php:62]: Won't this always be true? I thought `getenv` returned a string

Comment (hawx) [reply service-front/app/config/autoload/envs.global.php:62]: And even if it did return a boolean

```php
php > echo ((false ?: true) ? "yes" : "no");
yes
```

Comment (cooperaj) [reply service-front/app/config/autoload/envs.global.php:62]: getenv returns a string or null. the ?: syntax is shorthand for `getenv('COOKIE_SECURE') ? getenv('COOKIE_SECURE') : true`

Comment (cooperaj) [reply service-front/app/config/autoload/envs.global.php:62]: Good spot.

File: service-front/app/config/autoload/envs.global.php [hunk 2]
```diff
@@ -58,7 +58,8 @@
         // Indicates that the cookie should only be transmitted over a
         // secure HTTPS connection from the client. When set to TRUE, the
         // cookie will only be set if a secure connection exists.
-        'cookie_secure' => false,
+//        'cookie_secure' => false,
```
Comment (cooperaj) [inline service-front/app/config/autoload/envs.global.php:61]: don't need this comment

File: service-front/app/config/autoload/envs.global.php (hunk 1 above)
Comment (cooperaj) [inline service-front/app/config/autoload/envs.global.php:62]:
```
'cookie_secure' => getenv('COOKIE_SECURE') === 'false' ? false : true,
```

should do it.

Comment (hawx) [reply service-front/app/config/autoload/envs.global.php:62]: I placed too much trust in the documented type `getenv ( string $varname [, bool $local_only = FALSE ] ) : string`

but in fact

> Returns the value of the environment variable varname, or FALSE if the environment variable varname does not exist.

Comment (SeemaMenon) [reply service-front/app/config/autoload/envs.global.php]: amended

File: service-front/app/config/autoload/envs.global.php (hunk 2 above)
Comment (SeemaMenon) [reply service-front/app/config/autoload/envs.global.php]: removed

Comment (SeemaMenon) [reply service-front/app/config/autoload/envs.global.php]: done

File: service-front/app/config/autoload/envs.global.php (hunk 1 above)
Comment (SeemaMenon) [reply service-front/app/config/autoload/envs.global.php]: applied

File: features/steps/viewer.py [hunk 3]
```diff
@@ -3,7 +3,6 @@
 import json
 from modules import get_frontend_url
```
Comment (williamfalconeruk) [inline features/steps/viewer.py:5]: mising import for environment variables. `import os`. see: https://stackoverflow.com/questions/4906977/how-to-access-environment-variable-values.

File: service-front/app/features/context/UI/AccountContext.php [hunk 4]
```diff
@@ -2598,5 +2598,36 @@ public function iAmShownANotFoundError()
...
+    /**
+     * @Then /^I want to ensure cookie attributes are set$/
+     */
+    public function iWantToEnsureCookieAttributesAreSet()
+    {
+        $session = $this->ui->getSession();
+
+        // retrieving response headers:
+        $cookies = $session->getResponseHeaders()['set-cookie'];
+        foreach ($cookies as $value) {
```
Comment (cooperaj) [inline service-front/app/features/context/UI/AccountContext.php:2628]: We'll only want to check the session cookie. Once the cookie consent form goes in there will be another cookie we send and this test will start failing.

File: service-front/app/features/context/UI/AccountContext.php [hunk 5]
```diff
@@ -2598,5 +2598,36 @@ public function iAmShownANotFoundError()
...
+    {
+        // Not needed for this context
+    }
+
+    /**
+     * @When /^I access the use a lasting power of attorney web page$/
+     */
+    public function iAccessTheUseALastingPowerOfAttorneyWebPage()
+    {
+        $this->ui->visit('https://use.lastingpowerofattorney.opg.service.justice.gov.uk/');
```
Comment (cooperaj) [inline service-front/app/features/context/UI/AccountContext.php:2615]: Wait, why are we visiting the live service?

File: features/steps/viewer.py (hunk 3 above)
Comment (SeemaMenon) [reply features/steps/viewer.py:5]: Yeah, is the reason tests fails. Added the import

File: service-front/app/features/context/UI/AccountContext.php (hunk 5 above)
Comment (SeemaMenon) [reply service-front/app/features/context/UI/AccountContext.php:2615]: Good spot, was meant to be local. Amended!

Comment (SeemaMenon) [reply service-front/app/features/context/UI/AccountContext.php:2615]: Good spot...was a test change. Amended now.

File: features/steps/viewer.py (hunk 3 above)
Comment (SeemaMenon) [reply features/steps/viewer.py:5]: fixed...was the reason test was failing too.

File: service-front/app/features/context/UI/AccountContext.php (hunk 4 above)
Comment (SeemaMenon) [reply service-front/app/features/context/UI/AccountContext.php:2629]: Fix added...review pls

File: service-front/app/features/context/UI/AccountContext.php [hunk 6]
```diff
@@ -2598,5 +2599,40 @@ public function iAmShownANotFoundError()
...
+        $session = $this->ui->getSession();
+
+        // retrieving response headers:
+        $cookies = $session->getResponseHeaders()['set-cookie'];
+        foreach ($cookies as $value) {
+            if (strstr($value,'session')) {
+                assertContains('secure', $value);
+                assertContains('httponly', $value);
+            } else {
+                throw new Exception('session not found');
```
Comment (cooperaj) [inline service-front/app/features/context/UI/AccountContext.php:2634]: Maybe make this error message a little more verbose

**Pull request merged**